### ISSUE TITLE
[java] interface declaration and call arguments mismatch

### ISSUE THREAD

**Issue by ilyash-b** [jcoglan/canopy#22]
Hi.

Failing grammar excerpt:

```
true_           <- "true"  %makeTrue
```

Grammar workaround:

```
true_           <- "true" sub_elements_workaround %makeTrue
sub_elements_workaround <- "%DUMMY%"?
```

Action implementation:
```
    public TrueTreeNode makeTrue(String input, int start, int end, List<TreeNode> elements) {
```

Without `sub_elements_workaround` compilation fails because the interface (`public interface Actions`) says the action method parameters include input, start, end and elements while the generated parser calls the action only with input, start and end.

**Comment by LMLB** [comment]
I believe the problem is [this line](https://github.com/jcoglan/canopy/blob/e1bcc7a9318c2156c313ec4e3e78d1c5e6480607/src/builders/java.js#L404).

Changing that line to `args.push(elements || this.null_());` would make it pass `null` instead (which is pretty much the same behavior as the other three languages).

**Comment by jcoglan** [comment]
I believe this is fixed in 79360be64ada4249dfcb4b509d51c485ef8a6ddd and then refactored in 053e8773d122b1fad11c075873f420bdfdc94d43, which applies this change across all language backends and removes the ambiguity of whether or not the `elements` argument is optional.

**Comment by jcoglan** [comment]
This should now be fixed as part of the v0.4 release; see https://blog.jcoglan.com/2022/04/01/announcing-canopy-0-4/
